src/oci.c

```diff
@@ -595,6 +769,14 @@ cc_oci_create_container_workload (struct cc_oci_config *config)
 		goto out;
 	}
 
+	quoted_workload_cmd = g_shell_quote (workload_cmdline);
+	sg_command = g_strdup_printf ("sg %s -c %s",
```

Comment:
The bad news is that the `sg` command is not always available. For example, the popular `busybox` docker image only provides `su`.

We have 2 main options here:

- Require `sg` if `gid != 0` and we can't find the `sg` command, fail the request and thus fail the container start.
  That's a safe approach but not very friendly since the user won't know if we can honour their request until the last moment.

- Create a static binary (must be static) that we make available in the workload chroot and have that handle uid, gid and cwd changes before having it exec the required workload. I already have code to do this fwiw so will send that through. This overall approach isn't idea either, but may be the best pragmatic solution since:
  - it does not require the workload image contain particular userspace binaries.
  - it guarantees that we can always honour the request.